### PR TITLE
[singleproject] add a new @(MauiAsset) item group

### PR DESCRIPTION
**Pull request by jonathanpeppers**
### Description of Change ###

For Maui's "single project" concept, we need a new item group for raw
files:

    <MauiAsset Include="Raw\index.html" />

This would do the following for each platform:

* Android - add an `@(AndroidAsset)` to the `.apk`
* iOS/MacCatalyst - add a `@(Content)` to the `.app`/`.ipa`
* WinUI - add a `@(Content)` to the `Assets/` folder

To test this, I added an `index.html` file to
`Maui.Controls.Sample.SingleProject.csproj` and used it via:

    <WebView Source="index.html" />

The only issue is on Android we need to load
`file:///android_asset/index.html`. I modified the Android
`WebViewRenderer` to do:

    if (!url.StartsWith('/') && !Uri.IsWellFormedUriString(url, UriKind.Absolute))
    {
        url = AssetBaseUrl + url;
    }

So for the following examples:

* `index.html` -> `file:///android_asset/index.html`
* `./index.html` -> `file:///android_asset/./index.html`
* `https://google.com` -> no change
* `/some/full/path/on/disk/index.html` -> no change, this is why we
  need the explicit `/` check.

I think this is improved `WebView` behavior, in general.

Down the road, we could make a helper API that can either return a
`System.Uri` or a `System.IO.Stream` in C# for all platforms.

Other changes:

* Updated `README.md` files that described a workaround for
  `--no-restore` that is no longer needed.
* `Maui.Controls.Sample.SingleProject.csproj` now actually uses `.svg`
  files for its app icons.
* `Maui.Controls.Sample.SingleProject.csproj` uses `.xaml` now.

### Additions made ###

* Adds `@(MauiAsset)` item group.

image

### PR Checklist ###

- [ ] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

### CODE REVIEW

**Comment by jonathanpeppers** [comment]
I was showing this to Jon Pryor, and we think we can remove the need for the `{OnPlatform}` in the example above.

If you look at:

https://github.com/dotnet/maui/blob/10803a338ef6f5ce12f7adaffceb85a52e6dcafa/src/Compatibility/Core/src/Android/Renderers/WebViewRenderer.cs#L44-L51

If `url` is not an absolute URL, it can't possibly load on Android.

So we could auto-prepend this if the URL isn't absolute:

https://github.com/dotnet/maui/blob/10803a338ef6f5ce12f7adaffceb85a52e6dcafa/src/Compatibility/Core/src/Android/Renderers/WebViewRenderer.cs#L18

Which would make `Source="index.html"` work on Android without any custom logic.

I'll go ahead and add this today, but let me know if anyone sees an issue with this.

**Comment by Redth** [comment]
@jonathanpeppers we can't assume we always want to load a URI from a file though, it might just be a https:// url that's used in some cases.  Maybe the web view should just have a `LoadUrlFromAssets(..)` method which prepends the uri base correctly for each platform...

**Comment by jonathanpeppers** [comment]
@Redth the idea is if it has `https://` in the URL it is absolute, and so we would only append `file:///android_asset/` if it was a path like `index.html`.

I'm thinking of doing:
```csharp
if (!Uri.IsWellFormedUriString (url, UriKind.Absolute))
{
	url = AssetBaseUrl + url;
}
```
If `url` is `"index.html"`, it wouldn't work anyway, so doesn't hurt to attempt using `AssetBaseUrl`.

I will test this a bit, though.

File: src/Compatibility/Core/src/Android/Renderers/WebViewRenderer.cs [hunk 1]
```diff
@@ -46,6 +46,11 @@ void LoadUrl(string url, bool fireNavigatingCanceled)
 			if (!fireNavigatingCanceled || !SendNavigatingCanceled(url))
 			{
 				_eventState = WebNavigationEvent.NewPage;
+				if (url != null && !url.StartsWith('/') && !Uri.IsWellFormedUriString(url, UriKind.Absolute))
+				{
+					// URLs like "index.html" can't possibly load, so try "file:///android_asset/index.html"
+					url = AssetBaseUrl + url;
+				}
```
Comment (jonathanpeppers) [inline src/Compatibility/Core/src/Android/Renderers/WebViewRenderer.cs:53]: @Redth here is the new logic. Does someone else need to review?

I added example URLs I tested in the PR description.

**Review by Redth — APPROVED** [review]

**Pull request merged**